admin/commands/execution/stop_at_height.go

```diff
@@ -0,0 +1,82 @@
+package execution
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/admin"
+	"github.com/onflow/flow-go/admin/commands"
+)
+
+var _ commands.AdminCommand = (*StopAtHeightCommand)(nil)
+
+// StopAtHeightCommand will send a signal to engine to stop/crash EN
+// at given height
+type StopAtHeightCommand struct {
+	height *atomic.Uint64
+	crash  *atomic.Bool
+}
+
+func NewStopAtHeightCommand(height *atomic.Uint64, crash *atomic.Bool) *StopAtHeightCommand {
+	return &StopAtHeightCommand{
+		height: height,
+		crash:  crash,
+	}
+}
+
+type StopAtHeightReq struct {
+	height uint64
+	crash  bool
+}
+
+func (s *StopAtHeightCommand) Handler(ctx context.Context, req *admin.CommandRequest) (interface{}, error) {
+	sah := req.ValidatorData.(StopAtHeightReq)
+
+	oldHeight := s.height.Swap(sah.height)
+	oldCrash := s.crash.Swap(sah.crash)
```

Comment:
wondering if these two lines should be in the same critical section, if we care about atomicity here? like if A sends "1, true", then B sends "2, false", it may still happen that the received value is "1, false" etc. but if we don't expect concurrent command issuing, then it is fine doing so here.

Reply:
Its a good question, I considered it, but I doubt we need to care about race condition here